src/server/rdb_load.cc

```diff
@@ -738,9 +738,29 @@ void RdbLoaderBase::OpaqueObjLoader::CreateHMap(const LoadTrace* ltrace) {
 }
 
 void RdbLoaderBase::OpaqueObjLoader::CreateList(const LoadTrace* ltrace) {
-  quicklist* ql =
-      quicklistNew(GetFlag(FLAGS_list_max_listpack_size), GetFlag(FLAGS_list_compress_depth));
-  auto cleanup = absl::Cleanup([&] { quicklistRelease(ql); });
+  quicklist* ql;
+  if (config_.append) {
+    if (pv_->ObjType() != OBJ_LIST) {
```

Comment:
This code is repetitive. Can be refactored into `bool EnsureObjEnctoding(const PrimeValue& pv)` const`

Reply:
sounds good